synapse/handlers/identity.py

```diff
@@ -61,21 +61,78 @@ def _should_trust_id_server(self, id_server):
                 return False
         return True
 
+    def _extract_items_from_creds_dict(self, creds, require_access_token=True):
+        """
+        Retrieve entries from a "credentials" dictionary
+
+        Args:
+            creds (dict[str, str]): Dictionary of credentials that contain the following keys:
+                * client_secret|clientSecret: A unique secret str provided by the client
+                * id_server|idServer: the domain of the identity server to query
+                * id_access_token: The access token to authenticate to the identity
+                    server with.
+            require_access_token: Whether id_access_token is a required key in creds
+
+        Returns:
+            tuple(str, str, str|None): A tuple containing the client_secret, the id_server,
+                and optionally the id_access_token values.
+        """
+        client_secret = creds.get("client_secret") or creds.get("clientSecret")
+        if not client_secret:
+            raise SynapseError(
+                400, "No client_secret in creds", errcode=Codes.MISSING_PARAM
+            )
+
+        id_server = creds.get("id_server") or creds.get("idServer")
+        if not id_server:
+            raise SynapseError(
+                400, "No id_server in creds", errcode=Codes.MISSING_PARAM
+            )
+
+        id_access_token = creds.get("id_access_token")
```

Comment:
What's this? Do we really want to be giving the HS the IS access token? Aren't we using open ID?

Reply:
As far as I understand it `id_access_token` is a one-time-use auth token that the client receives from the Identity Server. The client is expected to hand this to the HS for proxying.

That process is described [here](https://github.com/matrix-org/matrix-doc/pull/1961/files#diff-46763da5dcada9c6bd410e646df8390bR19), which MSC2140 [links to](https://github.com/matrix-org/matrix-doc/pull/2140/files#diff-c03a26de5ac40fb532de19cb7fc2aaf7R89).